### ISSUE TITLE
XSS in bibtexbrowser.php via wp-publications Plugin

### ISSUE THREAD

**Issue by Zeynalxan** [monperrus/wp-publications#4]
### **Overview**  
A **Stored Cross-Site Scripting (XSS)** vulnerability was identified in the **wp-publications WordPress plugin** through its integration with the `bibtexbrowser.php` script. By creating a `.bib` file with a **malicious name**, the XSS payload is triggered **automatically on the default page**, without needing to pass the file name explicitly as a parameter.

---

### **Vulnerability Details**  
- **Vulnerable Plugin:** wp-publications  
- **Affected File:** `bibtexbrowser.php`  
- **Repository:** [https://github.com/monperrus/wp-publications](https://github.com/monperrus/wp-publications)  
- **Vulnerability Type:** Stored XSS  
- **Impact:** High (Unauthenticated remote code execution in the user’s browser)  

---

### **Steps to Reproduce**  

1. **Install the Plugin:**  
   Install the wp-publications plugin on a WordPress site and ensure that the `bibtexbrowser.php` is accessible.

2. **Create a Maliciously Named File:**  
   On the server, run the following command:

   ```bash
   touch "<img src=x onerror=alert('XSS')>.bib"
   ```

3. **Trigger the XSS:**  
   Open the **default page** at:  
   ```
   http://<your-website>/wp-content/plugins/wp-publications/bibtexbrowser.php?frameset&bib=
   ```

4. **Result:**  
   The **XSS payload** is executed, and the following alert box pops up:  
   ```
   alert('XSS')
   ```

---

### **Proof of Concept (PoC)**  

- **Malicious File Name:**  
  ```bash
  touch "<img src=x onerror=alert('XSS')>.bib"
  ```

- **Trigger URL:**  
  ```
  http://<your-website>/wp-content/plugins/wp-publications/bibtexbrowser.php?frameset&bib=
  ```

- **Observed Behavior:**  
  When accessing the page, the JavaScript alert `XSS` is triggered immediately.

---

### **Business Impact**  
This vulnerability can lead to:  
- **Session Hijacking:** Attacker can steal session cookies.
- **Malware Injection:** Users can be tricked into downloading malicious files.
- **Phishing Attacks:** Fake login prompts or redirects can be embedded via XSS.
- **Service Reputation Damage:** Unchecked vulnerabilities damage trust in the platform.

---

### **Root Cause**  
The **`bibtexbrowser.php`** script **does not properly sanitize file names**, allowing JavaScript code to execute when these filenames are displayed on the page.

---

### **Conclusion**  
This Stored XSS vulnerability demonstrates a critical flaw that can be exploited by attackers to run arbitrary JavaScript in the context of users' browsers. Immediate mitigation is recommended to prevent malicious exploitation.

---

**Comment by monperrus** [comment]
fixed by ce00bcb041b0a9c122fb68adfebe138d8b6087c0

**Comment by monperrus** [comment]
thanks for the bug report
